backend/src/server_process_helper.py

```diff
@@ -47,21 +47,28 @@ def __init__(self, flags: list[str]):
         self._stop_event = threading.Event()
 
         # Create a separate thread to read and print the output of the subprocess
-        threading.Thread(
+        self.t1 = threading.Thread(
             target=self._read_output,
             daemon=True,
             name="output reader",
-        ).start()
+        )
+        self.t1.daemon = True
+        self.t1.start()
 
     def close(self):
+        logger.info("Closing worker process...")
         self._stop_event.set()
-        self._process.terminate()
-        self._process.kill()
+        if hasattr(self, "_process"):
+            self._process.terminate()
+            self._process.kill()
+            del self._process
+        if hasattr(self, "t1"):
+            del self.t1
 
     def _read_output(self):
         if self._process.stdout is None:
             return
-        for line in self._process.stdout:
+        for line in iter(self._process.stdout):
```

Comment:
Why?

Reply:
This was one of the 500 things I was trying, and I forgot to remove it